### ISSUE TITLE
Make fetching workspaces `deleted` query param consistent

### ISSUE THREAD

**Issue by Emyrk** [coder/coder#2176]
Currently if you attempt to fetch a deleted workspace, if it is deleted you need to include `?deleted=true` query parameter. We also have a `include_deleted` query param in the case that it will return regardless if it is deleted or not.

If you fetch workspace**s**, then the `?deleted` query param only shows deleted workspace OR it shows active workspaces. It will never show a mixed set of deleted and active. From a UI perspective, this is kinda strange, as you'd assume there are cases you want the mixed set.

# Proposal

We should keep only the `?deleted` query param and treat it as follows:

- `deleted=""` : Show both deleted and active
 - This is the default case
- `deleted="false": Show only active
- `deleted="true"`: Show only deleted

The empty string applies no filters and returns the mixed set. We should add a `Deleted` field to `Workspace` data type for the user to see the workspace status.

# Reasoning

## Fetching single

When fetching a single workspace, we already have the `id` or the `name`. We want to know if the workspace exists regardless of it's state. They should check the `Deleted` field to know if it's active or not. I assume we might even allow the workspace to be brought back to life in the future?

If they want to only see active/deleted they can use the `deleted` query param

## Fetching multiple

In our other filters we treat empty string as "do not apply the filter", so the same case goes here. We just need that `Deleted` field for the user to know the status.

**Comment by Kira-Pilot** [comment]
This is a great writeup; thanks for adding it. I think your reasoning is solid and an implementation as described above would definitely make our lives easier.

**Comment by Emyrk** [comment]
@misskniss @tjcran A good one for the prettier issue: https://github.com/coder/coder/issues/1939

**Comment by Emyrk** [comment]
@ketang that is how it can be done on the FE. On the backend `owner:me` is translated into `?owner=me`

**Comment by ketang** [comment]
We should use that query syntax in API calls as well.  It shouldn't be something we do on the front end only.  I see value in this for general API use but also the CLI.  `coder workspaces search` (or `list` takes arguments potato batata).

**Comment by Emyrk** [comment]
@ketang I support passing the query directly.

**Comment by Emyrk** [comment]
After chatting with @f0ssel @Kira-Pilot we decided not to allow showing deleted workspaces in the workspace list view. It's not needed and can actually cause more issues than it's worth.

I made WorkspaceByID and WorkspaceByOwnerAndName match `include_deleted` and fixed which deleted workspace is returned in `WorkspaceByOwnerAndName`. There was a bug if you deleted the same workspace name more than once.


This will suffice for community mvp.
https://github.com/coder/coder/pull/2214
